### DISCUSSION TITLE
How best to handle alternatively named Gemfile or symlink in project?

### DISCUSSION THREAD

**Discussion by blischalk** [renovatebot/renovate#9074] [Request Help] [answered]
I'm working on a project where the `Gemfile` and `Gemfile.lock` are symlinks to `Gemfile-current` and `Gemfile-current.lock` respectively. Currently it does not seem that Renovate follows the symlink of the Gemfile. It also doesn't seem to pick up on the Gemfiles named with a suffix. I have tried using `"matchFiles": ["Gemfile-current", "Gemfile-current.lock"]` as part of the `config.js` but this doesn't seem to get Renovate to pick them up.

Are there configurations that can be added to get Renovate to follow the symlink or pick up alternatively named Gemfiles?

**Comment by rarkins** [comment, marked as answer]
`matchFiles` is for `packageRules` - not what you want.

Most likely it's this:

```json
{
  "bundler": {
    "fileMatch": ["Gemfile-current"]
  }
}
```

**Reply by blischalk** [reply]
Thank you @rarkins ! That config worked.

**Reply by blischalk** [reply]
@rarkins After looking at the MR's that Renovate opened with that fileMatch config you suggested, I noticed that the lock files are not being updated as well while using the `update-lockfile` rangeStrategy:

```
{repository: 'project/reponame', rangeStrategy: 'update-lockfile', "bundler": { "fileMatch": ["Gemfile-current"] }}
```

I proceeded to try adding `Gemfile-current.lock` into the `fileMatch` config as well but this doesn't seem to solve the problem:

```
{repository: 'project/reponame', rangeStrategy: 'update-lockfile', "bundler": { "fileMatch": ["Gemfile-current", "Gemfile-current.lock"] }}
```

Instead, when adding `Gemfile-current.lock` to the fileMatch strategy it seems to actually break things when Renovate tries to update merge requests, throwing errors such as the following:

```
ERROR: Error ensuring PR: Response code 400 (Bad Request) (repository=project/repo, branch=branch-fix-gem-blah-1.x)
       "err": {
         "name": "HTTPError",
         "timings": {
           "start": 1616012166034,
           "socket": 1616012166034,
           "lookup": 1616012166035,
           "connect": 1616012166038,
           "secureConnect": 1616012166039,
           "upload": 1616012166039,
           "response": 1616012166043,
           "end": 1616012166044,
           "phases": {
             "wait": 0,
             "dns": 1,
             "tcp": 3,
             "tls": 1,
             "request": 0,
             "firstByte": 4,
             "download": 1,
             "total": 10
           }
         },
         "message": "Response code 400 (Bad Request)",
         "stack": "HTTPError: Response code 400 (Bad Request)\n    at Request.<anonymous> (/workdir/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)",
         "options": {
           "headers": {
             "user-agent": "https://github.com/renovatebot/renovate",
             "accept-encoding": "gzip, deflate",
             "private-token": "**redacted**",
             "accept": "application/json",
             "content-type": "application/json",
             "content-length": "1455"
           },
           "url": "https://gitlab.host.here/merge_requests/999/notes",
           "username": "",
           "password": "",
           "method": "POST",
           "http2": false
         },
         "response": {
           "statusCode": 400,
           "statusMessage": "Bad Request",
           "body": "Bad Request",
           "headers": {
             "server": "nginx",
             "date": "Wed, 17 Mar 2021 20:16:06 GMT",
             "content-type": "text/plain",
             "content-length": "11",
             "connection": "close",
             "x-request-id": "xxx",
             "x-runtime": "0.000955"
           },
           "httpVersion": "1.1"
         }
       }
```

Any suggestions on how to configure Renovate to also pick up on and update the alternatively named lock file?

**Reply by rarkins** [reply]
Please revert rangeStrategy back to the default ("range"), and reset `fileMatch` to the default too, and if the problem persists. Assuming so, see if you can create a minimal reproduction in a public repo so I can take a look.

**Reply by blischalk** [reply]
@rarkins after reverting rangeStrategy back to default and fileMatch, the above error message goes away but it goes back to the situation of not picking up on the alternatively named `Gemfile-current` / `Gemfile-current.lock`.

I will try to see if I can create a minimal reproduction in a public repo.

**Reply by blischalk** [reply]
Following up on this. I did still encounter an issue after reverting but things worked fine when creating a minimal reproduction. 

The minimal test case was created [here](https://github.com/blischalk/renovate-bundler-test) and we can see in the MR's that Renovate opened that it did in fact work with the alternative named Gemfiles and also adjusted the lock files.

Looking back into the issue further and reviewing the errors in the logs I was receiving I found:

```
"stderr": "Could not find gem 'Ruby\u0000 (~> 2.5.8.0)' in the local ruby installation.\nThe source contains 'Ruby\u0000' at: 2.7.2.137\n",
         "message": "Command failed: bundle lock --update aws-sdk-sns\nCould not find gem 'Ruby\u0000 (~> 2.5.8.0)' in the local ruby installation.\nThe source contains 'Ruby\u0000' at: 2.7.2.137\n"
```

which seems highly likely to be related to this issue:

https://github.com/renovatebot/renovate/issues/3074

The Gemfile for the project has the Ruby version set with `ruby '2.5.8'`.

Since this issue seems unrelated to the different named Gemfiles I still think it is safe to say that `"bundler": { "fileMatch": ["Gemfile-current"] }` does work as expected for handling differently named Gemfiles and to address my other problem, I will need to look into how Renovate ended up dealing with declaring Ruby versions in the Gemfile.
